x-pack/plugins/enterprise_search/public/applications/app_search/components/create_engine/create_engine_logic.test.ts

```diff
@@ -0,0 +1,135 @@
+        CreateEngineLogic.actions.onCreateEngineSuccess();
+      });
+
+      it('should set a success message', () => {
+        expect(setQueuedSuccessMessage).toHaveBeenCalledWith(CREATE_ENGINE_SUCCESS_MESSAGE);
+      });
+
+      it('should navigate the user to the engine page', () => {
+        const enginePath = generatePath(ENGINE_PATH, { engineName: CreateEngineLogic.values.name });
+        expect(navigateToUrl).toHaveBeenCalledWith(enginePath);
+      });
+    });
+
+    describe('setLanguage', () => {
+      const newLanguage = 'English';
+
+      beforeAll(() => {
+        mount();
+        CreateEngineLogic.actions.setLanguage(newLanguage);
+      });
+
+      describe('language', () => {
+        it('should be set to provided value', () => {
+          expect(CreateEngineLogic.values.language).toEqual(newLanguage);
+        });
+      });
+    });
+
+    describe('setRawName', () => {
+      const newName = 'Name__With#$&*%Special--Characters';
+      const sanitizedNewName = formatApiName(newName);
+
+      beforeAll(() => {
+        mount();
+        CreateEngineLogic.actions.setRawName(newName);
+      });
+
+      describe('rawName', () => {
+        it('should be set to provided value', () => {
+          expect(CreateEngineLogic.values.rawName).toEqual(newName);
+        });
+      });
+
+      describe('name', () => {
+        it('should be set to a sanitized value', () => {
+          expect(CreateEngineLogic.values.name).toEqual(sanitizedNewName);
+        });
+      });
+    });
+
+    describe('submitEngine', () => {
+      beforeAll(() => {
+        mount();
+        // setup local state
+        CreateEngineLogic.actions.setLanguage('English');
+        CreateEngineLogic.actions.setRawName('test');
+      });
+
+      it('POSTS to /api/app_search/engines', () => {
+        const body = JSON.stringify({
+          name: CreateEngineLogic.values.name,
+          language: CreateEngineLogic.values.language,
+        });
+        CreateEngineLogic.actions.submitEngine();
+        expect(http.post).toHaveBeenCalledWith('/api/app_search/engines', { body });
+      });
+
+      it('calls onCreateEngineSuccess on valid submission', async () => {
+        jest.spyOn(CreateEngineLogic.actions, 'onCreateEngineSuccess'); // .mockImplementation();
+        const promise = (http.post as jest.Mock).mockReturnValueOnce({});
+        await CreateEngineLogic.actions.submitEngine();
+        await promise;
+        expect(CreateEngineLogic.actions.onCreateEngineSuccess).toHaveBeenCalledTimes(1);
+      });
+
+      it('calls flashAPIErrors on API Error', async () => {
+        const promise = (http.post as jest.Mock).mockReturnValueOnce(
+          Promise.reject({
+            body: {
+              statusCode: 400,
+              error: 'Bad Request',
+              message: 'Invalid request payload JSON format',
+            },
+          })
+        );
+        await CreateEngineLogic.actions.submitEngine();
+        await promise;
```

Comment:
On a side note, I'm surprised this doesn't error out as is on `await promise`, since it's a rejected promise. Cases like this were the reason we created that `expectAsyncError` helper. 😵

Reply:
Ahh great question - I wonder if that's because of the auto-mock PR (#88494) where we now automatically mock `flashAPIError()` as a flat jest.fn(), so it doesn't re-throw errors? 🤔